lib/resque.rb

```diff
@@ -255,6 +295,30 @@ def after_pause=(block)
     register_hook(:after_pause, block)
   end
 
+  def before_reserve(&block)
+    block ? register_hook(:before_reserve, block) : hooks(:before_reserve)
+  end
+
+  def before_reserve=(block)
+    register_hook(:before_reserve, block)
+  end
```

Comment:
FEATURE: `reservable_queues`

This hook is used to determine which queues should be checked for work. The block will be called once for each of the worker's assigned queues; raising `Resque::DontReserve` will skip that queue. This is used to support per-queue pausing in the monolith.

This could be PRed to master, but is only used by the `blocking_reserve` code currently.